File: life-assisitant-api/internal/controller/user_category.go
// Package controller 习惯/待办分类控制器（20260922 新增，见 06 §3.3）
// 规范：controller 只做参数解析 + 调 service + 返回响应
package controller

import (
	"github.com/gogf/gf/v2/net/ghttp"

	"github.com/life-assistant/api/internal/consts/ecode"
	"github.com/life-assistant/api/internal/model/dto"
	"github.com/life-assistant/api/internal/response"
	"github.com/life-assistant/api/internal/service"
)

// UserCategoryController 习惯/待办分类控制器
type UserCategoryController struct{}

var UserCategory = &UserCategoryController{}

// List GET /user-categories?domain=habit|task
// ⚠️ 该 domain 首次请求会自动播种内置分类（4 / 6 条）
func (c *UserCategoryController) List(r *ghttp.Request) {
	out, err := service.UserCategory().List(r.Context(), r.Get("domain").String())
	if err != nil {
		writeError(r, err)
		return
	}
	response.Success(r, out)
}

// Create POST /user-categories
func (c *UserCategoryController) Create(r *ghttp.Request) {
	var req dto.CreateUserCategoryReq
	if err := r.Parse(&req); err != nil {
		response.Error(r, ecode.ValidationFailed, err)
		return
	}
	out, err := service.UserCategory().Create(r.Context(), &req)
	if err != nil {
		writeError(r, err)
		return
	}
	response.Created(r, dto.UserCategoryItemResp{Item: *out})
}

// Update PATCH /user-categories/:id
func (c *UserCategoryController) Update(r *ghttp.Request) {
	id := r.Get("id").String()
	if id == "" {
		response.Error(r, ecode.MissingRequiredField)
		return
	}
	var req dto.UpdateUserCategoryReq
	if err := r.Parse(&req); err != nil {
		response.Error(r, ecode.ValidationFailed, err)
		return
	}
	out, err := service.UserCategory().Update(r.Context(), id, &req)
	if err != nil {
		writeError(r, err)
		return
	}
	response.Success(r, dto.UserCategoryItemResp{Item: *out})
}

// Delete DELETE /user-categories/:id → 204（软删三件套）
func (c *UserCategoryController) Delete(r *ghttp.Request) {
	id := r.Get("id").String()
	if id == "" {
		response.Error(r, ecode.MissingRequiredField)
		return
	}
	if err := service.UserCategory().Delete(r.Context(), id); err != nil {
		writeError(r, err)
		return
	}
	response.NoContent(r)
}

File: life-assisitant-ui-desktop/src/api/user-category.ts
// ============================================================================
// 用户分类 API —— 习惯 / 待办（spec-20260922-v2/06 §3）
//
// SYNC-FROM-MOBILE: life-assisitant-ui-mobile/src/api/user-category.ts
//
// ⚠️ 与收支分类（`/finance/categories`，两级树）是**两套实体**：
//    这里一级平铺、按 domain=habit|task 分域；创建 / 更新返回 `{item}`
//    包裹（与财务控制器同口径），删除 204 无响应体（request 拦截器解包）。
// ============================================================================
import request from './request'
import type {
  CreateUserCategoryReq,
  ListUserCategoriesQuery,
  ListUserCategoriesResp,
  UpdateUserCategoryReq,
  UserCategoryItemResp,
} from './types'

export const userCategoryApi = {
  /** 按域拉取分类列表（06 §3.1）。seeded = 后端本次是否懒播种 */
  list(params: ListUserCategoriesQuery) {
    return request.get<unknown, ListUserCategoriesResp>('/user-categories', { params })
  },

  /** 创建分类（06 §3.2）。id 由后端生成（uc_<domain>_<random>），返回 {item} */
  create(data: CreateUserCategoryReq) {
    return request.post<unknown, UserCategoryItemResp>('/user-categories', data)
  },

  /** 更新分类（06 §3.3，PATCH /:id）。domain 不可改，返回 {item} */
  update(id: string, data: UpdateUserCategoryReq) {
    return request.patch<unknown, UserCategoryItemResp>(`/user-categories/${id}`, data)
  },

  /** 删除分类（软删除，204 无响应体） */
  remove(id: string) {
    return request.delete(`/user-categories/${id}`)
  },
}
